### PR TITLE
Use the AWS_REGION set in the config and not hardcoded ones

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
This PR change the hardcoded AWS Region to use the one we set in the config, if that is not set then we use a default one.
Also changed the RDS creation to get the availabilities zones available in the region and also not use hardcoded.

This also addresses the mock generation to not have the user path in the mocks, so we can add a check to validate if the mocks are out of sync

#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/MM-27183

#### Release Note

```release-note
Use the AWS_REGION set in the config and not hardcoded ones
```

### CODE REVIEW

File: internal/tools/aws/rds.go [hunk 1]
```diff
@@ -93,12 +93,13 @@ func (a *Client) rdsEnsureDBClusterCreated(awsID, vpcID, username, password, kms
...
+		return err
+	}
+
 	input := &rds.CreateDBClusterInput{
-		AvailabilityZones: []*string{
-			aws.String("us-east-1a"),
-			aws.String("us-east-1b"),
-			aws.String("us-east-1c"),
-		},
+		AvailabilityZones:     azs,
```
Comment (cpanato) [inline internal/tools/aws/rds.go:102]: Here, I dont know if we want to cap to 2-3 or want to add all available

**Comment by gabrieljackson** [comment]
Sorry, looks like the Postgres PR changes conflicted with this.

**Comment by cpanato** [comment]
@gabrieljackson thanks :D will rebase later not today 😂

**Comment by cpanato** [comment]
@gabrieljackson @gigawhitlocks PTAL

File: internal/tools/terraform/plan.go [hunk 2]
```diff
@@ -27,11 +30,16 @@ func (c *Cmd) Init(remoteKey string) error {
...
+	awsRegion := os.Getenv("AWS_REGION")
+	if awsRegion == "" {
+		awsRegion = toolsAWS.DefaultAWSRegion
+	}
+
 	_, _, err = c.run(
 		"init",
 		arg("backend-config", fmt.Sprintf("bucket=%s", c.remoteStateBucket)),
 		arg("backend-config", fmt.Sprintf("key=%s/%s", remoteStateDirectory, remoteKey)),
-		arg("backend-config", "region=us-east-1"),
```
Comment (angeloskyratzakos) [inline internal/tools/terraform/plan.go:34]: @cpanato maybe to leave that one at `us-east-1` as the `-backend-config=bucket=mattermost-kops-state-test` exists in us-east-1 or we can have a different approach on using different bucket on other regions

Comment (cpanato) [reply internal/tools/terraform/plan.go:34]: i think we need to have a different bucket for other regions as well to not mess up

File: internal/tools/aws/rds.go (hunk 1 above)
Comment (gabrieljackson) [reply internal/tools/aws/rds.go:118]: This should be fine, I would think.

File: internal/tools/aws/client_test.go [hunk 3]
```diff
@@ -139,7 +139,7 @@ func NewAWSTestSuite(t *testing.T) *AWSTestSuite {
 		RDSParamGroupCluster: "mattermost-provisioner-rds-cluster-pg",
 		RDSParamGroup:        "mattermost-provisioner-rds-pg",
 		RDSClusterID:         "rds-cluster-multitenant-09d44077df9934f96-97670d43",
-		RDSAvailabilityZones: []string{"us-east-1a", "us-east-1b", "us-east-1c"},
+		RDSAvailabilityZones: []string{"us-honk-1a", "us-honk-1b"},
```
Comment (gigawhitlocks) [inline internal/tools/aws/client_test.go:142]: What availability zone is this?

Comment (cpanato) [reply internal/tools/aws/client_test.go:142]: /honk

**Pull request merged**